namerd/iface/interpreter-thrift/src/main/scala/io/buoyant/namerd/iface/ByteBufferSerializers.scala

```diff
@@ -11,11 +11,14 @@ object ByteBufferSerializers {
   def path(path: Seq[ByteBuffer]): String =
     path.map(utf8).mkString("/", "/", "")
 
-  def stamp(bb: ByteBuffer): String =
-    bb.duplicate().getLong.toString
+  def stamp(bb: ByteBuffer): String = {
+    if (bb.array().length > 0) bb.duplicate().getLong.toString else ""
```

Comment:
probably better not to assume that the BB is array-backed.  we can use `bb.remaining()` to get number of available bytes in the buffer.

Also, longs are 8 bytes so we should ensure that there are at least 8 bytes remaining: https://docs.oracle.com/javase/7/docs/api/java/nio/ByteBuffer.html#getLong()

Reply:
TIL